ibis/expr/operations/relations.py

```diff
@@ -329,6 +329,14 @@ def schema(self):
         return sch.Schema(names, types)
 
 
+def _add_alias(op: ops.Value | ops.TableNode):
```

Comment:
Why is it required? All value operations must be named already.

Reply:
Even if we want to force aliasing here why not for TableColumns then?

Reply:
We would also need to restrict the selections fields to only allow aliases instead of generic values.

Reply:
This is required because subselects generate expressions without aliases in some backends. This means any subsequent references to the corresponding column in such subselects will produce invalid code, because there's no predictable name generated to reference.

Reply:
We don't need to do this for `TableColumn`s because those have predictable names.

Reply:
> We would also need to restrict the selections fields to only allow aliases instead of generic values.

Why? Simply making this the case, as I am doing here should be sufficient.

Reply:
We also cannot just blindly alias everything, since that would start to generate invalid code like

```
SELECT *
FROM t
WHERE x AS thing = 1
```